release/models/bgp/openconfig-bgp-policy.yang

```diff
@@ -661,6 +670,297 @@ module openconfig-bgp-policy {
+                  description "State for BGP color extended community
+                  defined by RFC9012";
+                  config false;
+                  uses ext-community-color-config;
+              }
+          }
+        }
+      }
+  }
+
+grouping ext-community-asn-type-config {
+  description "Grouping for asn-type extended community config.";
+  leaf asn {
+      type oc-inet:as-number;
+      description "Autonomous system number.";
+  }
+
+  leaf value {
+      type    uint16;
+      description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-ip-type-config {
+  description "Grouping for ip-type extended community config.";
+  leaf asn {
+    type oc-inet:ip-address;
+    description "IPv4 Address for the community.";
+  }
+
+  leaf value {
+    type    uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-link-bw-config {
+  description "Grouping for link bandwidth extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf bps {
+    type uint64;
+    description "Bandwidth in bits per second.";
+  }
+}
+
+grouping ext-community-route-target-config {
+  description "Grouping for route-target extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-target-ip-config {
+  description "Grouping for route-target-ip extended community
+  config.";
+  leaf ip-address {
+    type oc-inet:ip-address;
+    description "IPv4 address.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-origin-config {
+  description "Grouping for origin extended community config.";
+  leaf asn {
+    type oc-inet:as-number;
+    description "Autonomous system number.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-route-origin-ip-config {
+  description "Grouping for route-origin-ip extended community config.";
+  leaf ip-address {
+    type oc-inet:ip-address;
+    description "IPv4 address.";
+  }
+
+  leaf value {
+    type uint16;
+    description "Assigned value for this community.";
+  }
+}
+
+grouping ext-community-color-config {
+  description "Grouping for color extended community config.";
+  leaf match-type {
+    type oc-bgp-types:bgp-color-match-type;
+    description "Specifies the match criteria between BGP NH and
+    SR policy endpoint";
+  }
+
+  leaf value {
+    type uint32;
+    description "Assigned value for this community.";
+  }
+}
+
+  grouping ext-community-member-config {
+    description
+    "Configuration data for typed BGP extended communities.";
+
+    leaf index {
+      type uint16;
+      mandatory true;
```

Comment:
`mandatory`  is a design pattern not taken up for other list keys across OC models to date - it is only softly implied by way of the key being a leafref to one of the config/state nodes but now this would enforce the non-leafref'd node to be mandatory (in this case state)

Would suggest consistency before all else

Both YANG 1.0/1.1 also state that the mandatory statement on key leafs are ignored.  This would also apply here since this node is the ultimate leafref

Reply:
Thanks for this feedback.  Good point on the consistency, and the yang standard, I will remove this.